core/evalir.ml

```diff
@@ -656,24 +657,24 @@ struct
             (Lens.Value.sort lens1)
             (Lens.Value.sort lens2) ~on
           |> Lens_errors.unpack_sort_join_result ~die:(eval_error "%s") in
-        apply_cont cont env (`Lens (Value.LensJoin {left; right; on; del_left; del_right; sort}))
+        apply_cont cont env (`Lens (db1, Value.LensJoin {left; right; on; del_left; del_right; sort}))
     | LensCheck (lens, _typ) ->
```

Comment:
Here it looks like lens checking is a no-op.  Is that correct or is the plan to put more code here?

Reply:
This is currently a no-op which is just a slight divergence from the theoretical behaviour. In theory the dynamic checks happen during evaluation of the `LensCheck` expression, in practice the dynamic checks happen during lens construction (but can still require a `LensCheck` expression to statically acknowledge runtime behaviour).

Reply:
You don't want to eliminate `lenscheck`, because it is used to differentiate between statically checkable lenses and dynamically checkable lenses. So the difference between the first and third example you had given is that the third example might be fine, if <expression> can be typechecked statically, then it would compile and run. If it could not be typechecked statically, then the first example would run and the third would refuse to compile if used by a `lensput` expression.

Reply:
Ah, OK, I remember now.  So maybe what I want is for lenscheck to be translated away after typechecking but that would be a separate issue.

Reply:
I think it would be better for it to eventually be the deferred location for catching errors. I'm abusing this PR to also add a comment in the code and will merge it after the CI passes.